nodebuilder/header/header.go

```diff
@@ -4,32 +4,78 @@ import (
 	"context"
 
 	"github.com/celestiaorg/celestia-node/header"
+	libhead "github.com/celestiaorg/celestia-node/libs/header"
 )
 
 // Module exposes the functionality needed for querying headers from the network.
 // Any method signature changed here needs to also be changed in the API struct.
 //
 //go:generate mockgen -destination=mocks/api.go -package=mocks . Module
 type Module interface {
+	// Head returns the ExtendedHeader of the chain head.
+	Head(context.Context) (*header.ExtendedHeader, error)
+
+	// Get returns the header of the given hash from the node's header store.
+	Get(ctx context.Context, hash libhead.Hash) (*header.ExtendedHeader, error)
+	// GetVerifiedRangeByHeight returns the given range [from:to) of ExtendedHeaders
+	// from the node's header store and verifies that the returned headers are
+	// adjacent to each other.
+	GetVerifiedRangeByHeight(
+		ctx context.Context,
+		from *header.ExtendedHeader,
+		to uint64,
+	) ([]*header.ExtendedHeader, error)
 	// GetByHeight returns the ExtendedHeader at the given height, blocking
 	// until header has been processed by the store or context deadline is exceeded.
 	GetByHeight(context.Context, uint64) (*header.ExtendedHeader, error)
-	// Head returns the ExtendedHeader of the chain head.
-	Head(context.Context) (*header.ExtendedHeader, error)
+
 	// IsSyncing returns the status of sync
 	IsSyncing(context.Context) bool
+	// WaitSync blocks until the header Syncer is synced to network head.
+	WaitSync(ctx context.Context) error
+	// SyncHead provides the Syncer's view of the current network head.
+	SyncHead(ctx context.Context) (*header.ExtendedHeader, error)
+
+	// Subscribe creates long-living Subscription for validated ExtendedHeaders.
+	// Multiple Subscriptions can be created.
+	Subscribe(context.Context) (libhead.Subscription[*header.ExtendedHeader], error)
 }
 
 // API is a wrapper around Module for the RPC.
 // TODO(@distractedm1nd): These structs need to be autogenerated.
 type API struct {
 	Internal struct {
+		Head func(context.Context) (*header.ExtendedHeader, error) `perm:"public"`
+		Get  func(
+			ctx context.Context,
+			hash libhead.Hash,
+		) (*header.ExtendedHeader, error) `perm:"public"`
+		GetVerifiedRangeByHeight func(
+			context.Context,
+			*header.ExtendedHeader,
+			uint64,
+		) ([]*header.ExtendedHeader, error) `perm:"public"`
 		GetByHeight func(context.Context, uint64) (*header.ExtendedHeader, error) `perm:"public"`
-		Head        func(context.Context) (*header.ExtendedHeader, error)         `perm:"public"`
-		IsSyncing   func(context.Context) bool                                    `perm:"public"`
+		IsSyncing   func(context.Context) bool                                    `perm:"read"`
+		WaitSync    func(ctx context.Context) error                               `perm:"read"`
+		SyncHead    func(ctx context.Context) (*header.ExtendedHeader, error)     `perm:"read"`
```

Comment:
SyncHead(NetworkHead) should be public IMO, while Head(LocalHead) should be read instead

Reply:
done